.github/workflows/tests.yml

```diff
@@ -0,0 +1,27 @@
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  test:
+    strategy:
+      matrix:
+        ruby: [2.5, 2.6, 2.7]
+    runs-on: ubuntu-latest
+    name: Test on Ruby ${{ matrix.ruby }}
+    steps:
+      - uses: actions/checkout@v2
+      - uses: ruby/setup-ruby@v1
+        with:
+          ruby-version: ${{ matrix.ruby }}
+      - uses: actions/cache@v1
+        with:
+          path: vendor/bundle
+          key: bundle-${{ matrix.ruby }}-${{ hashFiles('**/*.gemspec') }}
+          restore-keys: bundle-${{ matrix.ruby }}
+      - run: bundle install --jobs 4 --retry 3 --path vendor/bundle
```

Comment:
Does `nproc` work on Github? I find some allusions to it timing out. There's a workaround as well: https://github.community/t/number-of-cores-as-env-context-variable/17396

Then again, gems are cached, so it's not big deal even if 4 is a suboptimal number of jobs.

Reply:
Also check the deprecation warning:

> [DEPRECATED] The `--path` flag is deprecated because it relies on being remembered across bundler invocations, which bundler will no longer do in future versions. Instead please use `bundle config set path 'vendor/bundle'`, and stop using this flag

Reply:
Thanks, I'll replace the `--path` argument.

I'll also reduce the number of jobs to `2` because apparently the Github-hosted virtual machines all have two cores (https://docs.github.com/en/free-pro-team@latest/actions/reference/specifications-for-github-hosted-runners).